### PR TITLE
Dataset ops v1.1

### PR DESCRIPTION
**Pull request by regit**
Update of #5754 fixing some tests.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Fix a non void function not returning a value in case of error.
- Rebase on latest master

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: doc/userguide/rules/datasets.rst [hunk 1]
```diff
@@ -169,6 +169,50 @@ type
 data
   Data to remove in serialized form (base64 for string, hex notation for md5/sha256)
 
+dataset-remove
```
Comment (victorjulien) [inline doc/userguide/rules/datasets.rst:172]: name does not match example below (`dataset-clear`)

File: doc/userguide/rules/datasets.rst [hunk 2]
```diff
@@ -169,6 +169,50 @@ type
...
+Syntax::
+
+    dataset-clear <set name> <set type>
+
+set name
+  Name of an already defined dataset
+type
+  Data type: string, md5, sha256
+
+dataset-exist
```
Comment (victorjulien) [inline doc/userguide/rules/datasets.rst:187]: I'd be in favor of a different name. Something like `dataset-match` or `dataset-lookup` or something similar.

**Review by victorjulien — COMMENTED** [review]
Wonder if/how we could automate testing, to add coverage for the new code.

File: doc/userguide/rules/datasets.rst (hunk 2 above)
Comment (regit) [reply doc/userguide/rules/datasets.rst:187]: dataset-lookup will match the hash table code?

**Pull request closed without merging**